libp2p-websocket/src/lib.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2017 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+#![recursion_limit = "512"]
+
+// TODO: use this once stable ; for now we just copy-paste the content of the README.md
+//#![doc(include = "../README.md")]
+
+//! Implementation of the libp2p `Transport` trait for Websockets.
+//!
+//! See the documentation of `swarm` and of libp2p in general to learn how to use the `Transport`
+//! trait.
+//!
+//! This library is used in a different way depending on whether you are compiling for emscripten
+//! or for a different operating system.
+//!
+//! # Emscripten
+//!
+//! On emscripten, you can create a `WsConfig` object with `WsConfig::new()`. It can then be used
+//! as a transport.
+//!
+//! Listening on a websockets multiaddress isn't supported on emscripten. Dialing a multiaddress
+//! which uses `ws` on top of TCP/IP will automatically use the `XMLHttpRequest` Javascript object.
+//!
+//! ```ignore
```

Comment:
Why `ignore` and not `no_run`?

Reply:
This demonstrates the emscripten code, but the doctests are compiled for x86 and hence will fail to compile.